### PR TITLE
Combined PR for #619, #621, #622

### PR DESCRIPTION
**Pull request by msprotz**
After my unfortunate merge of the alignment changes in krml, the only way to get a successful build is to have all three cryspen PRs together. They can still be reviewed separately but this is the one we'll have to merge.

### CODE REVIEW

**Review by R1kM — APPROVED** [review]
I did the review on the other PRs, as long as changes (if any) to #619 are propagated to this PR, this looks good.

**Comment by msprotz** [comment]
@polubelova need a second review on this, but the changes are being reviewed individually in their respective PRs. it's just that we need all three combined for a green. thanks!

**Pull request merged**
